backend/core/exceptions.py

```
@@ -0,0 +1,128 @@
+
+class ParkingAnalysisError(Exception):
+    """Custom exception for parking analysis errors"""
+    def __init__(self, message: str, error_type: str = "analysis_error"):
+        self.message = message
+        self.error_type = error_type
+        super().__init__(self.message)
+
+
+class ServiceUnavailableError(Exception):
+    """Custom exception for service unavailability"""
+    def __init__(self, service_name: str, message: str = None):
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)